05-onchain.md

```diff
@@ -325,252 +328,265 @@ transaction output.
 
 ## Penalty Transactions Weight Calculation
 
 There are three different scripts for penalty transactions, with the following
-witnesses weight (details of the computation are in
+witnesses weights (details of weight computation are in
 [Appendix A](#appendix-a-expected-weights)):
 
     to_local_penalty_witness: 160 bytes
     offered_htlc_penalty_witness: 243 bytes
     accepted_htlc_penalty_witness: 249 bytes
 
-The penalty txinput itself takes 41 bytes, which has a weight of 164, making the
-weight of each input:
+The penalty *txinput* itself takes up 41 bytes and has a weight of 164 bytes,
+which results in the following weights for each input:
 
     to_local_penalty_input_weight: 324 bytes
     offered_htlc_penalty_input_weight: 407 bytes
     accepted_htlc_penalty_input_weight: 413 bytes
 
-The rest of the penalty transaction takes 4+1+1+8+1+34+4=53 bytes of non-witness
-data, assuming that it has a pay-to-witness-script-hash (the largest standard output
-script), in addition to a 2-byte witness header.
+The rest of the penalty transaction takes up 4+1+1+8+1+34+4=53 bytes of
+non-witness data: assuming it has a pay-to-witness-script-hash (the largest
+standard output script), in addition to a 2-byte witness header.
 
-In addition to outputs being swept under as penalty, the node MAY also sweep the
-`to_remote` output of the commitment transaction (e.g. to reduce the total
-amount paid in fees). Doing so requires the inclusion of a p2wpkh witness and
-additional txinput, resulting in an additional 108 + 164 = 272 bytes.
+In addition to spending these outputs, a penalty transaction may optionally
+spend the commitment transaction's `to_remote` output (e.g. to reduce the total
+amount paid in fees). Doing so requires the inclusion of a P2WPKH witness and an
+additional *txinput*, resulting in an additional 108 + 164 = 272 bytes.
 
-In a worst case scenario, a node holds only incoming HTLCs, and the HTLC-timeout
-transactions are not published, which forces the node to spend from the commitment
-transaction.
+In the worst case scenario, the node holds only incoming HTLCs, and the
+HTLC-timeout transactions are not published, which forces the node to spend from
+the commitment transaction.
 
-With a maximum standard weight of 400000, the maximum number of HTLCs that can
-be swept in a single transaction:
+With a maximum standard weight of 400000 bytes, the maximum number of HTLCs that
+can be swept in a single transaction is as follows:
 
     max_num_htlcs = (400000 - 324 - 272 - (4 * 53) - 2) / 413 = 966
 
-This allows 483 HTLCs in each direction (with both `to_local` and
-`to_remote` outputs) to still be resolved in a single penalty transaction.
-Note that even if the `to_remote` output is not swept, the resulting
-`max_num_htlcs` is 967, which yields the same unidirectional limit of 483 HTLCs.
+Thus, 483 bidirectional (containing both `to_local` and
+`to_remote` outputs) HTLCs can be resolved in a single penalty transaction.
```

Comment:
Might be a little clearer if the parenthetical note was after HTLCs rather than before. I think it's still just as accurate.

Reply:
good call